### PR TITLE
Niganesh/tests for dao classes

### PR DESCRIPTION
**Pull request by nithyaganeshng**
Tests for dao classes. 
- It uses in memory database of rooms for all db operations
- It includes positive and edge case tests

**Type of change:**
- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            data = o
+            latch.countDown()
+            removeObserver(this)
+        }
+    }
+    this.observeForever(observer)
+    latch.await(time, TimeUnit.SECONDS)
+    this.removeObserver(observer)
+    return data as T
+}
```
Comment (AndiMiko) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt:24]: that's a sweet helper 👍

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.lifecycle.Observer
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+fun <T> LiveData<T>.getOrAwaitValue(time: Long = 2): T {
+    var data: T? = null
+    val latch = CountDownLatch(1)
+    val observer = object : Observer<T> {
+        override fun onChanged(o: T?) {
+            data = o
```
Comment (AndiMiko) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt:15]: nit, one letter variable `o`.

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            latch.countDown()
+            removeObserver(this)
```
Comment (AndiMiko) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt:17]: nit, if I read this correctly this call is unnecessary since the observer is always removed after `latch.await` anyhow?

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        }
+    }
+    this.observeForever(observer)
+    latch.await(time, TimeUnit.SECONDS)
+    this.removeObserver(observer)
+    return data as T
```
Comment (AndiMiko) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt:23]: this crashes when `await` cancels after `time` runs out. Why not return a nullable value?

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/dao/PortableIdentityCardDaoInstrumentedTest.kt [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+        val portableIdentityCard2 = createPortableIdentityCard(1)
+        runBlocking {
+            portableIdentityCardDao.insert(portableIdentityCard1)
+            Assertions.assertThatThrownBy { runBlocking { portableIdentityCardDao.insert(portableIdentityCard2) } }
+                .isInstanceOf(android.database.sqlite.SQLiteConstraintException::class.java)
+            val actualCard = portableIdentityCardDao.getAllCards().getOrAwaitValue()
+            assertThat(actualCard.size).isEqualTo(1)
+            assertThat(actualCard).contains(portableIdentityCard1)
+        }
+    }
```
Comment (AndiMiko) [inline sdk/src/androidTest/java/com/microsoft/did/sdk/repository/dao/PortableIdentityCardDaoInstrumentedTest.kt:93]: The tests in this class are actually testing multiple cases and examine edge cases. That looks useful. 👍

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt (hunk 2 above)
Comment (nithyaganeshng) [reply sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt]: modified it to be meaningful variable name

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt (hunk 3 above)
Comment (nithyaganeshng) [reply sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt]: true. thought this would remove the observer immediately but not sure what happens when it is tried to remove twice. have removed this line.

File: sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt (hunk 4 above)
Comment (nithyaganeshng) [reply sdk/src/androidTest/java/com/microsoft/did/sdk/repository/LiveDataTestUtil.kt]: modified it to return nullable values here and asserted/handled null values in test.
